backend/app/main.py

```diff
@@ -1083,6 +1085,28 @@ async def get_projects(
     ).all()
     return gerrydb_views
 
+@app.get("/api/document/{document_id}/thumbnail", status_code=status.HTTP_200_OK)
+async def get_thumbnail(
+    *,
+    document_id: str,
+    session: Session = Depends(get_session),
+):
+    try:
+        img_io = fetch_thumbnail(session, document_id)
+        return StreamingResponse(content=img_io, media_type="image/png")
+    except:
+        return RedirectResponse(url="/home-megaphone.png")
+
+
+@app.put("/api/document/{document_id}/put_thumbnail", status_code=status.HTTP_200_OK)
+async def put_thumbnail(
```

Comment:
H: This endpoint is pretty heavy. I tried it with a texas map and saw about 15 seconds of blocking time. Would [FastAPI background tasks](https://fastapi.tiangolo.com/tutorial/background-tasks/#add-the-background-task) or better yet a lambda or cron job make sense here? CC @raphaellaude

Reply:
I'll switch to background task